python/pyspark/sql/pandas/types.py

```diff
@@ -86,8 +86,15 @@ def to_arrow_type(dt: DataType) -> "pa.DataType":
     elif type(dt) == DayTimeIntervalType:
         arrow_type = pa.duration("us")
     elif type(dt) == ArrayType:
-        if type(dt.elementType) in [StructType, TimestampType]:
+        if type(dt.elementType) == TimestampType:
             raise TypeError("Unsupported type in conversion to Arrow: " + str(dt))
+        elif type(dt.elementType) == StructType:
+            if LooseVersion(pa.__version__) < LooseVersion("2.0.0"):
+                raise TypeError(
+                    "Array of StructType is only supported with pyarrow 2.0.0 and above"
```

Comment:
Hi, @LucaCanali I think I've found a case where Array of StructType doesn't work properly:

**In:**
```python
df = spark.createDataFrame(
  [
    ("a", [("b", False), ("c", True)]),
  ]
).toDF("c1", "c2")
df.toPandas()
```

**Out:**
```python
  c1                                                 c2
0  a  [{'_1': 'b', '_2': False}, {'_1': 'c', '_2': T...
```

**Expected:**
```python
  c1                       c2
0  a  [(b, False), (c, True)]
```


I roughly suspect that this is maybe an internal table conversion issue from PyArrow side but not very sure about this, so could you check this out when you find some time??

Thanks in advance!

Reply:
Hi, @itholic thank you for reporting this.
I have tried to reproduce the reported issue in my environment, but I get the expected result instead (I use Python 3.11, Spark 3.5.1, pyarrow 11). Can you please confirm that you still have this issue?